pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/service/actvityinstancebuilder/form/block/FormBlockCreatorHelper.java

```diff
@@ -65,6 +73,33 @@ ComponentBlock createComponentBlock(AIBuilderContext ctx, ComponentBlockDef comp
         return new ComponentBlock(formComponent);
     }
 
+    TabularBlock createTabularBlock(AIBuilderContext ctx, TabularBlockDef tabularBlockDef) {
+        final List<List<Question>> rows = new ArrayList<>();
+        for (int row = 0; row < tabularBlockDef.getRowsCount(); row++) {
+            final List<Question> questionsRow = new ArrayList<>();
+            for (int column = 0; column < tabularBlockDef.getColumnsCount(); column++) {
+                final QuestionDef questionDef = tabularBlockDef.get(row, column);
+                if (questionDef == null) {
+                    questionsRow.add(null);
+                    continue;
+                }
```

Comment:
In what cases should we be anticipating `null` values in the row, and are there ways this can be avoided?

Reply:
I would think if one of the cells will have NO question. For ex: 3 columns and one of the rows will have only 2 questions. Not sure if we have this support though. Will let dimitry confirm

Reply:
Sampath is absolutely right. I assume we might have empty cells.